### ISSUE TITLE
No images were found to optimize error when using external images & custom loader

### ISSUE THREAD

**Issue by johanhenriksson** [dc7290/next-export-optimize-images#408]
**Describe the bug**
As soon as I add a custom loader, `next-export-optimize-images` no longer detects any images to download. As soon as it is removed, everything works fine again. It does not matter what the loader actually does, even a function simply returning the input `src` will stop it from working.

It's perhaps worth mentioning that I'm using external images from a CDN. It has been whitelisted in `next.config.js`, and as previously mentioned it works fine without the loader.

Any help or ideas would be greatly appreciated, this plugin looks like exactly what we need :) Thanks!

**To Reproduce**
1. Add a custom loader to an image component:
   ```ts
   import NextImage, { ImageLoaderProps, ImageProps } from 'next/image'

   const dummyLoader = ({ src }: ImageLoaderProps) => src
   export const Image = (props: ImageProps) => <NextImage loader={dummyLoader} {...props} />
   ```
2. Build & export
3. No images are downloaded. next-export-optimize-images exits with the message `No images were found to optimize.
`

**Expected behavior**
Images should be properly collected, downloaded and optimized even when a custom loader is provided.

**Additional information**
The entire site is pre-rendered using Nextjs Static Site Optimization.

**Comment by dc7290** [comment]
@johanhenriksson 

Thanks for issuing the Issue!

Can you tell me what patterns require setting up a loader and still optimizing the images at build time?
The pattern of using the loader, which was anticipated in advance, is when you want to use image optimization at build time overall, but use external images (Imgix, Cloudinary, etc.) in some areas.
It will be used in the following ways!
https://next-export-optimize-images.vercel.app/docs/examples#use-nextimage-separately-from-image-components-to-be-optimized-at-build-time

If you have any other use cases I'd love to know!

Thanks!

**Comment by johanhenriksson** [comment]
Yeah, I realized that after looking into the source code :) I think it makes sense. However I suggest adding some notes about the intent in the documentation, since it's not really clear. Before studying the source, it just looked like an example of how I could use custom loaders with the plugin.

As for the use case, I wanted to use some of the features provided by my external image API, like on-the fly cropping, and then have this plugin collect everything to be hosted on our own CDN (that lacks these features). I think some of this could be solved by simply exporting some of the internal components of the plugin and letting users compose their own Image component that leverages your `exportableLoader`. I thought about perhaps writing something that could wrap my custom loader with the `exportableLoader`, allowing modifications to the source URL before downloading the image. 

Either way, I have decided to just go with the external CDN for now, since this plugin does not seem to solve the original problem I wanted to solve - which was to embed `blurDataURL` for external images. But it seems to me that even using this, the `blurDataURL` would end up being a link to a small image and not actually a base64-encoded data url embedded directly into the HTML?
